hilti/toolchain/include/ast/operators/generic.h

```diff
@@ -17,29 +23,150 @@
 
 namespace hilti::operator_ {
 
+BEGIN_OPERATOR_CUSTOM(generic, Pack)
+    Type result(const hilti::node::Range<Expression>& ops) const {
+        if ( ops.empty() )
+            return type::DocOnly("<packable>");
+
+        return type::Bytes();
+    }
+
+    bool isLhs() const { return false; }
+    auto priority() const { return hilti::operator_::Priority::Normal; }
+
+    const std::vector<Operand>& operands() const {
+        static std::vector<Operand> _operands = {Operand{{}, type::Tuple(type::Wildcard())}};
+        return _operands;
+    }
+
+    void validate(const expression::ResolvedOperator& i, operator_::position_t p) const {
+        const auto args = i.op0().type().template as<type::Tuple>().elements();
```

Comment:
Honestly I'm not even sure what exactly prevents us from using per-type overloads here instead of a single generic operator; I indeed just adapted this from what `unpack` already does. It could be that the resolver is still not able to match against specific enum types (we tend to use wildcard enums elsewhere). I think it's ok for now.